nalgebra-sparse/src/csr.rs

```diff
@@ -596,6 +591,60 @@ impl<T> CsrMatrix<T> {
     }
 }
 
+#[cfg(feature = "serde-serialize")]
+mod serde_serialize {
+    use super::CsrMatrix;
+    use serde::{de, Deserialize, Deserializer, Serialize, Serializer};
+
+    #[derive(Serialize, Deserialize)]
+    struct CsrMatrixSerializationData<Indices, Values> {
```

Comment:
Would maybe be nice to have a comment that explains the rationale for using the intermediate type? For the sake of future maintainers it would be good to document this!